### PR TITLE
ttl: allow outbound foreign keys on TTL tables

### LINKED ISSUE #101225: sql: support outbound FKs on TTL tables
Outbound foreign keys are unaffected by rows being deleted from the TTL table itself and should be easier to support than inbound foreign keys.

Jira issue: CRDB-26871

Epic CRDB-26101

### PR DESCRIPTION
**Pull request by ecwall**
Fixes #101225

This change allows TTL tables to have outbound foreign keys. Outbound foreign
keys are unaffected by rows being deleted from the TTL table itself.

Release note (sql change): Allow outbound foreign keys on TTL tables.

### CODE REVIEW

File: pkg/sql/catalog/tabledesc/validate.go [hunk 1]
```diff
@@ -198,23 +198,14 @@ func (desc *wrapper) ValidateForwardReferences(
...
-			))
-		}
-		if len(desc.InboundForeignKeys()) > 0 {
-			vea.Report(unimplemented.NewWithIssuef(
-				76407,
-				`foreign keys to table with TTL %q are not permitted`,
-				desc.GetName(),
-			))
-		}
+	if desc.HasRowLevelTTL() && len(desc.InboundForeignKeys()) > 0 {
```
Comment (otan) [inline pkg/sql/catalog/tabledesc/validate.go:203]: any reason not to support all the cases if we're not doing strict filtering?

Comment (ecwall) [reply pkg/sql/catalog/tabledesc/validate.go:203]: I will handle it in a followup PR.

There is an issue with job SessionData not respecting `optDrivenFKCascadesClusterLimit` that I have to resolve https://github.com/cockroachdb/cockroach/issues/101265.

**Review by otan — APPROVED** [review]
probably want rafi to double check

**Review by rafiss — APPROVED** [review]
nice!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101236#-:-NSq9W6sEv7dIUcs0HMO:b-7m74ox)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

**Comment by ecwall** [comment]
bors r=otan,rafiss

**Pull request merged**